src/main/java/duke/TaskFactory.java

```diff
@@ -0,0 +1,54 @@
+package duke;
+
+import java.time.LocalDate;
+
+
+public class TaskFactory {
+    /**
+     * This is the method responsible for instantiating the right subtype of task.
+     * @param taskStr The string input from the user to be stored as a task
+     * @return Task A task object with the user's
+     * @exception IllegalFormatException Incorrect format for creating a task, prompts to correct format
+     * @exception EmptyDescriptionException If the description of a todo task is empty
+     * @exception InvalidCommandException If the taskStr doesn't match any task's format at all
+     */
+    public static Task createTask(String taskStr) throws IllegalFormatException, EmptyDescriptionException, InvalidCommandException {

```

Comment:
Perhaps a line break here could improve readability? Does this line have more than 120 chars?